File: PathoFroh/src/main/java/com/patho/main/action/dialog/diagnosis/CopyHistologicalRecordDialog.java
package com.patho.main.action.dialog.diagnosis;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Configurable;

import com.patho.main.action.dialog.AbstractDialog;
import com.patho.main.common.Dialog;
import com.patho.main.model.patient.Diagnosis;
import com.patho.main.service.TaskService;
import com.patho.main.util.dialogReturn.ReloadTaskEvent;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.Setter;

@Configurable
@Getter
@Setter
public class CopyHistologicalRecordDialog extends AbstractDialog {

	@Autowired
	@Getter(AccessLevel.NONE)
	@Setter(AccessLevel.NONE)
	private TaskService taskService;

	private Diagnosis diagnosis;

	public CopyHistologicalRecordDialog initAndPrepareBean(Diagnosis diagnosis) {
		if (initBean(diagnosis))
			prepareDialog();

		return this;
	}

	public boolean initBean(Diagnosis diagnosis) {
		setDiagnosis(diagnosis);
		super.initBean(task, Dialog.DIAGNOSIS_RECORD_OVERWRITE);
		return true;
	}

	public void copyHistologicalRecord(boolean overwrite) {
		taskService.copyHistologicalRecord(getDiagnosis(), overwrite);
	}

	public void hideDialog() {
		super.hideDialog(new ReloadTaskEvent());
	}

}

File: PathoFroh/src/main/kotlin/com/patho/main/dialog/diagnosis/CopyHistologicalRecordDialog.kt
package com.patho.main.dialog.diagnosis

import com.patho.main.common.Dialog
import com.patho.main.dialog.AbstractDialog_
import com.patho.main.model.patient.Diagnosis
import com.patho.main.service.DiagnosisService
import com.patho.main.util.event.dialog.TaskReloadEvent
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component

@Component()
@Scope(value = "session")
open class CopyHistologicalRecordDialog @Autowired constructor(
        private val diagnosisService: DiagnosisService) : AbstractDialog_(Dialog.DIAGNOSIS_RECORD_OVERWRITE) {

    open lateinit var diagnosis: Diagnosis

    open fun initAndPrepareBean(diagnosis: Diagnosis): CopyHistologicalRecordDialog {
        if (initBean(diagnosis))
            prepareDialog()
        return this
    }

    open fun initBean(diagnosis: Diagnosis): Boolean {
        this.diagnosis = diagnosis
        return super.initBean()
    }

    open fun copyAndHide(overwrite: Boolean) {
        hideDialog(TaskReloadEvent(diagnosisService.copyHistologicalRecord(diagnosis, overwrite)))
    }

}